001-time-format.md

```diff
@@ -0,0 +1,36 @@
+# Source Point Time Format
+
+## Summary
+
+Standardize the format of time information in source points to facilitate operations.
+
+## Motivation
+
+The time data of source points is stored according to its type:
+
+- Instant/snapshot points: hashmap with a single key "time";
+- Interval points: hashmap with keys "start" and "end".
+
+In order to operate on source points, it is required to either know in advance how the time information is stored or to [handle each different format](http://git.io/vf7GI).
+
+Operations such as sorting and selecting a subset of points could benefit from a standardized format.
+
+## Design
+
+The time information of source points can be reduced to a single timestamp plus some additional metadata:
+
+- Instant/snapshot points: just a timestamp, no additional metadata required;
+- Interval points: a single timestamp and a duration metadata.
+
+By giving a common name to the timestamp key of the time hashmap regardless of the source point type, it is possible to perform the operations described above without knowing the source point type in advance or handling all possible types.
```

Comment:
Gotcha. I don't mind making the type ("what kind of time is this?") implicit by virtue of the fields, since this is essentially going to be a primitive value, but if we do it that way then I think it has to be standardized, not left up to the providers (as I mention [here](https://github.com/uphex/uphex-rfcs/pull/2#issuecomment-96797651)).